### PR TITLE
ExpressionJIT simplify loop

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@mergifyio update

File: src/Interpreters/JIT/compileFunction.cpp [hunk 1]
```diff
@@ -94,12 +94,14 @@ static void compileFunction(llvm::Module & module, const IFunctionBase & functio
 
     /// Initialize loop
 
+    auto * end = llvm::BasicBlock::Create(b.getContext(), "end", func);
```
Comment (alexey-milovidov) [inline src/Interpreters/JIT/compileFunction.cpp:97]: Could you please write an explanation of the generated code in comments?
Something like a pseudocode of the generated constructions?

Comment (alexey-milovidov) [reply src/Interpreters/JIT/compileFunction.cpp:97]: Otherwise I just don't understand what's happening :crying_cat_face:

Comment (kitaisreal) [reply src/Interpreters/JIT/compileFunction.cpp:178]: Updated.

**Pull request merged**